commands/audit/jas/sast/sastscanner.go

```diff
@@ -22,41 +21,47 @@ const (
 type SastScanManager struct {
 	sastScannerResults []*sarif.Run
 	scanner            *jas.JasScanner
+	configFileName     string
+	resultsFileName    string
 }
 
-func RunSastScan(scanner *jas.JasScanner) (results []*sarif.Run, err error) {
-	sastScanManager := newSastScanManager(scanner)
-	log.Info("Running SAST scanning...")
-	if err = sastScanManager.scanner.Run(sastScanManager); err != nil {
+func RunSastScan(auditParallelRunner *utils.AuditParallelRunner, scanner *jas.JasScanner, extendedScanResults *utils.ExtendedScanResults, module jfrogappsconfig.Module, threadId int) (err error) {
+	var scannerTempDir string
+	if scannerTempDir, err = jas.CreateScannerTempDirectory(scanner, string(utils.Sast)); err != nil {
+		return
+	}
+	sastScanManager := newSastScanManager(scanner, scannerTempDir)
+	log.Info("[thread_id: " + strconv.Itoa(threadId) + "] Running SAST scanning...")
+	if err = sastScanManager.scanner.Run(sastScanManager, module); err != nil {
 		err = utils.ParseAnalyzerManagerError(utils.Sast, err)
 		return
 	}
 	if len(sastScanManager.sastScannerResults) > 0 {
-		log.Info("Found", utils.GetResultsLocationCount(sastScanManager.sastScannerResults...), "SAST vulnerabilities")
+		log.Info("[thread_id: "+strconv.Itoa(threadId)+"] Found", utils.GetResultsLocationCount(sastScanManager.sastScannerResults...), "SAST vulnerabilities")
 	}
-	results = sastScanManager.sastScannerResults
+	results := sastScanManager.sastScannerResults
+	auditParallelRunner.Mu.Lock()
+	extendedScanResults.SastScanResults = append(extendedScanResults.SastScanResults, results...)
+	auditParallelRunner.Mu.Unlock()
```

Comment:
this is not related to this logic, use it outside of this and return the result of the scan to be used outside.